### PR TITLE
chore(website): fix v6 beta post links to point to v6 site

### LINKED ISSUE #6755: Docs: v6 blog post links to main (v5) docs
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

Thanks @reintroducing for mentioning this in https://github.com/typescript-eslint/typescript-eslint/discussions/6014#discussioncomment-5411631!

https://typescript-eslint.io/blog/announcing-typescript-eslint-v6-beta includes links to docs, but those links are all on the main deployment of the website - which is still v5. We should change those links to point to the v6 preview of the site.

### Affected URL(s)

https://typescript-eslint.io/blog/announcing-typescript-eslint-v6-beta

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

- [x] Addresses an existing open issue: fixes #6755
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Fixes the links to point to https://v6--typescript-eslint.netlify.app/.

Also adds a `<!-- truncate -->` market while I'm here, so the blog post doesn't take up the entirety of https://typescript-eslint.io/blog. 😂

### CODE REVIEW

**Comment by kozlovvski** [comment]
> Also adds a  market while I'm here

Good catch with this one 😄

File: packages/website/blog/2023-03-13-announcing-typescript-eslint-v6-beta.md [hunk 1]
```diff
@@ -54,12 +56,12 @@ These are the changes that users of typescript-eslint -generally, any developer
...
 ### Reworked Configuration Names
 
-The biggest configuration change in typescript-eslint v6 is that we've reworked the names of our [provided user configuration files](https://typescript-eslint.io/linting/configs).
+The biggest configuration change in typescript-eslint v6 is that we've reworked the names of our [provided user configuration files](https://v6--typescript-eslint.netlify.app/linting/configs).
 typescript-eslint v5 provided three recommended configurations:
 
-- [`recommended`](https://typescript-eslint.io/linting/configs#recommended): Recommended rules for code correctness that you can drop in without additional configuration.
-- [`recommended-requiring-type-checking`](https://typescript-eslint.io/linting/configs#recommended-requiring-type-checking): Additional recommended rules that require type information.
-- [`strict`](https://typescript-eslint.io/linting/configs#strict): Additional strict rules that can also catch bugs but are more opinionated than recommended rules.
+- [`recommended`](https://v6--typescript-eslint.netlify.app/linting/configs#recommended): Recommended rules for code correctness that you can drop in without additional configuration.
```
[the remark this answers is not in the document]
Comment (JoshuaKGoldberg) [reply packages/website/blog/2023-03-13-announcing-typescript-eslint-v6-beta.md:62]: I don't think they can - if someone is on the main site while it's still v5, we'll want them to be taken to the v6 site.

**Comment by JoshuaKGoldberg** [comment]
I cherry-picked 6e4b987 into the `website` branch. It's on prod now. Thanks!

(normally we'd have to wait for our weekly release on Monday)

**Comment by reintroducing** [comment]
@JoshuaKGoldberg thank you for making these updates, much appreciated! 

I did want to add one comment here, just to provide a bit of color behind my original comments in the discussion. I'm a long time ESLint user who has managed multiple shareable configs over the years for different companies I've worked at that have been published to private npm registries. That being said, I am a novice TS user and had finally decided to implement TS in a large codebase so I wanted the linting capabilities that this package provides. I had noticed a few months ago that ESLint has the [new configuration docs](https://eslint.org/docs/latest/use/configure/configuration-files-new) and as part of this TS update I figured I'd also update the configuration to the new format (I thought I had read that this will become the default/suggested config format as of v9 so figured I'd get ahead of the game). 

I started off with updating the [JS recommended configs](https://www.npmjs.com/package/@eslint/js) which provide their installation instructions in the new format and moved on to the React configs. They do a REALLY nice job of providing a [legacy config](https://github.com/jsx-eslint/eslint-plugin-react#configuration-legacy-eslintrc) as well as the [new config](https://github.com/jsx-eslint/eslint-plugin-react#configuration-new-eslintconfigjs) section in their README which made it quite easy to set all of this up (plus I'm already familiar with these configs from prior usage). As a new TS/TS-eslint user, I'm not 100% sure how to best set this up in the new config system and what is/isn't needed in those new configs, so it would be amazing if something like this can be part of the v6 docs to help people like myself out who may possibly be starting with the newest version/config system in ESLint.

(Also now that I'm reading and gathering these links, [this was](https://github.com/jsx-eslint/eslint-plugin-react#configuration-new-eslintconfigjs) actually where I saw the note about the v9 default support, first paragraph).

**Pull request merged**